urllib3/response.py

```diff
@@ -150,7 +150,7 @@ def get_redirect_location(self):
         return False
 
     def release_conn(self):
-        if not self._pool or not self._connection:
+        if not self._pool or self._connection is None:
```

Comment:
I agree with you if the code is already relying on falsey values.
Also take a look at https://github.com/boyxuper/urllib3/blob/master/urllib3/response.py#L390 .
I'll update the pull request, please correct me if something wrong.